pkgs/applications/networking/p2p/qbittorrent/default.nix

```diff
@@ -1,44 +1,45 @@
-{ stdenv, fetchurl, pkgconfig, which
+{ stdenv, fetchFromGitHub, pkgconfig
 , boost, libtorrentRasterbar, qtbase, qttools, qtsvg
 , debugSupport ? false # Debugging
 , guiSupport ? true, dbus_libs ? null # GUI (disable to run headless)
 , webuiSupport ? true # WebUI
 }:
 
-assert guiSupport -> (dbus_libs != null);
-with stdenv.lib;
-
 stdenv.mkDerivation rec {
   name = "qbittorrent-${version}";
-  version = "4.1.0";
+  version = "4.1.1";
 
-  src = fetchurl {
-    url = "mirror://sourceforge/qbittorrent/${name}.tar.xz";
-    sha256 = "0fdr74sc31x421sb69vlgal1hxpccjxxk8hrrzz9f5bg4jv895pw";
+  src = fetchFromGitHub {
+    rev = "release-${version}";
+    owner = "qbittorrent";
+    repo = "qbittorrent";
+    sha256 = "09bf1jr2sfdps8cb154gjw7zhdcpsamhnfbgacdmkfyd7qgcbykf";
   };
 
-  nativeBuildInputs = [ pkgconfig which ];
+  nativeBuildInputs = [ pkgconfig ];
 
-  buildInputs = [ boost libtorrentRasterbar qtbase qttools qtsvg ]
-    ++ optional guiSupport dbus_libs;
+  buildInputs = [ boost libtorrentRasterbar qtbase qttools qtsvg
+    (if guiSupport then [ dbus_libs ] else [ ]) # D(esktop)-Bus depends on GUI support
```

Comment:
Please don't nest arrays in `buildInputs`. The old `buildInputs` are correct (and `with stdenv.lib` at the top is ok; alternatively you can add `lib` argument and write `lib.optional`).

Reply:
Yes, I included what previously was list concatenation into list.
Fixing.

Reply:
Since `optional` in the source code is
```nix
{
  optionals = cond: elems: if cond then elems else [];
}
```
I prefer to use `if then else` for uniformal look, when there are already `if then else` needed in the code.